File: app/src/main/java/com/novel/read/model/db/SearchListTable.java
package com.novel.read.model.db;


import androidx.annotation.Nullable;

import org.litepal.crud.LitePalSupport;

import java.io.Serializable;

/**
 * create by zlj on 2019/6/19
 * describe:
 */
public class SearchListTable extends LitePalSupport implements Serializable {

    private String key;
    private long saveTime;

    public String getKey() {
        return key == null ? "" : key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public long getSaveTime() {
        return saveTime;
    }

    public void setSaveTime(long saveTime) {
        this.saveTime = saveTime;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj != null && obj.toString().equals(key)) {
            return true;
        }
        return super.equals(obj);
    }
}

File: app/src/main/java/com/novel/read/model/db/SearchListTable.kt
package com.novel.read.model.db


import org.litepal.crud.LitePalSupport

import java.io.Serializable

/**
 * create by zlj on 2019/6/19
 * describe:
 */
class SearchListTable : LitePalSupport(), Serializable {

    var key: String = ""
    var saveTime: Long = 0


    override fun equals(other: Any?): Boolean {
        return if (other != null && other.toString() == key) {
            true
        } else super.equals(other)
    }

    override fun hashCode(): Int {
        var result = key.hashCode()
        result = 31 * result + saveTime.hashCode()
        return result
    }
}
